File: Source/WickedSick.Fayde.Client.Engine/Javascript/Controls/ToolTipService.js
/// <reference path="../Runtime/Nullstone.js"/>
/// <reference path="../Core/DependencyObject.js"/>
/// <reference path="../Core/UIElement.js"/>
/// CODE

(function (namespace) {
    var ToolTipService = Nullstone.Create("ToolTipService");

    ToolTipService.ToolTipProperty = DependencyProperty.RegisterAttached("ToolTip", function () { return Fayde.DependencyObject; }, ToolTipService);
    ToolTipService.PlacementTargetProperty = DependencyProperty.RegisterAttached("PlacementTarget", function () { return Fayde.UIElement; }, ToolTipService);

    namespace.ToolTipService = Nullstone.FinishCreate(ToolTipService);
})(Nullstone.Namespace("Fayde.Controls"));

File: Source/Fayde.Server/Controls/ToolTipService.cs
using Fayde.Core;
using Fayde.Xaml.Metadata;

namespace Fayde.Controls
{
    public class ToolTipService : DependencyObject
    {
        public static readonly AttachedPropertyDescription PlacementTargetProperty = AttachedPropertyDescription.Register("PlacementTarget", typeof(UIElement), typeof(ToolTipService));
        public static readonly AttachedPropertyDescription ToolTipProperty = AttachedPropertyDescription.Register("ToolTip", typeof(ToolTip), typeof(ToolTipService));
    }
}